.github/workflows/codecov.yml

```diff
@@ -0,0 +1,22 @@
+name: "Codecov"
+
+on: [push, pull_request]
+
+jobs:
+  code-coverage:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install lcov and run test
+      run: |
+        sudo apt-get install lcov
+        make lcov
```

Comment:
I believe this target is super slow since it only runs one test at a time (as opposed to the default which parallelizes a bunch of tests). It took 26m in your PR run. I don't want to delay small PRs that don't care much about coverage for this. I wonder if there is a way we can still parallelize the tests and/or find a way to really only run run the tests when we have to.

Reply:
Yeah, I found out it took a long time. 

The `make lcov` code in `src/Makefile`:
```shell
lcov:
	$(MAKE) gcov
	@(set -e; cd ..; ./runtest --clients 1)
	@geninfo -o valkey.info .
	@genhtml --legend -o lcov-html valkey.info
```

We can run `./runtest` without `--clients 1`，it can be accelerated to about 9m.